chihacknight/state-by-state-decarbonization
# Make State Details Graph Stick On Desktop (& Fix Homepage Electrification Lines)

## Overview

Move the large desktop layout to a consistent two column layout where a single stacked bar chart follows you along the page, animating between the different states we have setup. Additionally, the text fades in and out to accompany the graph, like so:


https://user-images.githubusercontent.com/3187531/163510318-d28a17e5-af6e-4f1f-aff2-29d52064d380.mp4

Closes #46

### Demo

See Overview.

### Notes

I used [react-scrollspy](https://www.npmjs.com/package/react-scrollspy) for this but it was recently deprecated. However since it is so light-weight I think that's okay.

## Testing Instructions

- Confirm the graph updates properly as you scroll on desktop (> 1200px wide)
- Confirm you see a proper alternating two column layout on tablet (768 - 1200px wide)
- Confirm you see a fully one column layout on mobile (< 768px wide)
